Update VTK to 9.5.0, ITK to 6.0.0 and Qt to 6.9

AlexyPellegrini opened the pull request:
This PR introduces the following changes:
- Update dependencies versions to VTK 9.5.0, ITK 6.0.0 and Qt 6.9
- Fixes some indirect issues related to dependencies update
- Clean VTK and ITK find package (now using components)
- Remove gengetopt from source, uses the one built by ITK instead
- Remove RTK from source, uses the one built by ITK instead
- Update CMake minimum required version. As of CMake 4.1, CMake < 3.10 is no longer supported.
- Add support for packaging the project using CPack (NSIS on windows, STGZ on Mac and Linux)

@LucasGandel

AlexyPellegrini committed:
ENH: Build with VTK 9.5+, ITK 6.0+ and Qt 6.0+

AlexyPellegrini committed:
ENH: Fix Linux build

AlexyPellegrini committed:
ENH: Add github workflow

AlexyPellegrini committed:
ENH: Use RTK from ITK remote modules

AlexyPellegrini committed:
ENH: remove gengetopt

AlexyPellegrini committed:
ENH: Update module list to build for VTK and ITK in CI

AlexyPellegrini committed:
ENH: Package binaries for ubuntu 22.04

Using ubuntu 22.04 grants better libc compatibility

AlexyPellegrini committed:
ENH: remove context menu from slice view

AlexyPellegrini committed:
ENH: fix suspicious down cast

AlexyPellegrini committed:
ENH: fix slice views update when selecting crop region

AlexyPellegrini committed:
ENH: fix histogram display when selecting one of multiple images

AlexyPellegrini committed:
ENH: Remove unused QtDesigner APIs

AlexyPellegrini committed:
ENH: Perform a static build of VV and its dependencies

AlexyPellegrini committed:
ENH: Fix blend image actor rendering with new OpenGL module

AlexyPellegrini committed:
BUG: Fix Qt warnings at exit

AlexyPellegrini committed:
BUG:  Fix keyboard interaction for slice views

SimonRit commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,157 @@
+    branches: [ master ]
+
+env:
+  INSTALL_PREFIX: "${{ github.workspace }}/install"
```
As far as I understand, I think all install steps can be removed as the install directory is not used anywhere.

SimonRit replied:
I had to specify `Qt6_ROOT=${INSTALL_PREFIX}` in the environment to let VTK find the compiled Qt instead of the system one.

SimonRit commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,157 @@
+    -DVTK_MODULE_USE_EXTERNAL_VTK_freetype=ON
+  VTK_DIR: "${{ github.workspace }}/vtk-build"
+  ITK_TAG: "v6.0a03"
+  ITK_MODULES: >-
```
You can add `-DRTK_BUILD_APPLICATIONS:BOOL=OFF`

AlexyPellegrini replied:
We need gengetopt from RTK as it may not be available on the system.

SimonRit replied:
Right, sorry for the noise!

AlexyPellegrini committed:
Workaround for ubuntu 22.04 QBackingStoreDefaultCompositor error

For some reason on ubuntu 22.04, QBackingStoreDefaultCompositor can not
compile shaders correctly until the mainwindow has been rendered once.
Let's use a short time to delay sequence loading.

AlexyPellegrini committed:
Enable SMP for VTK in CI

AlexyPellegrini committed:
Add QVTKOpenGLNativeWidget2 class

lesaintjerome commented:
Hello. Just a quick message to say that I seem to succeed in compiling vv on macos, apple silicon M4 chip, which is great. For some reason, I cannot open file from Ctrl+O or from the menu (but still to investigate. I'll raise an issue if confirmed). Dragging a file into vv works.
Thanks so much for having updated vv. I've felt like a naked soldier for the past months... Now I am powerful again !

AlexyPellegrini committed:
ENH: Adding JPEG as a static dependency

benpresles commented:
Hello. Does this merge mean that you no longer support ITK < 6, VTK < 9.5 and Qt < 6 ?

tbaudier commented:
Hello, not anymore
